### PR TITLE
TE-3514 fields changes when tabbing out

### PR DESCRIPTION
**Pull request by jperdiguerra**
JIRA Ticket: https://kaligo.atlassian.net/browse/TE-3514

**Issue:**
During selection, when ENTER and TAB are pressed in quick succession, the selected value changes back to the value at index 0.

**Cause:**
The current condition for selection with TAB allows either `ctrl.open = false` or `ctrl.open = true`.  This should be ok for most cases however, if the user pressed ENTER and TAB in quick succession, the selection will be triggered twice - one for the ENTER and another for TAB.  Since the index will be reset (to 0) from the ENTER selection, the active index when the TAB selection is run will now be 0, hence changing back the value to the first option.

**Fix:**
Update the condition such that, for single selection, a selection will not be triggered when `ctrl.open = false`.

Current condition `if (!ctrl.multiple || ctrl.open)`, this is truthy when:
`ctrl.multiple = false && ctrl.open = true`
`ctrl.multiple = false && ctrl.open = false`
`ctrl.multiple = true && ctrl.open = true`

Updated condition `if (ctrl.open)`, this is truthy when:
`ctrl.multiple = false && ctrl.open = true`
`ctrl.multiple = true && ctrl.open = true`

**How to test in TE:**
1. Apply the change in TE: `kaligo-web/node_modules/ui-select/dist/select.js`
2. Recompile webpack
3. Go to ANZ hotel/car landing page
4. Tab to the destination search box and search for a destination
5. When selecting, press enter then tab in quick succession
6. Validate that the highlighted destination is selected

**TODO:** 
Create PR for TE when this is ready (update in package.json - ui-select version).
`"ui-select": "git+https://github.com/Kaligo/ui-select#v0.19.8.4"`

### CODE REVIEW

File: dist/select.js [hunk 1]
```diff
@@ -862,7 +862,7 @@ uis.controller('uiSelectCtrl',
         }
         break;
       case KEY.TAB:
-        if (!ctrl.multiple || ctrl.open) ctrl.select(ctrl.items[ctrl.activeIndex], true);
+        if (ctrl.open) ctrl.select(ctrl.items[ctrl.activeIndex], true);
```
Comment (yidingww) [inline dist/select.js:865]: we shouldn't make the change here, this is `dist/` folder, which will be updated when you rebuild the project.

The correct place is line 585 of `uiSelectController.js`. Please revert the change here in dist folder.

Also, you need to build the library as well in this PR:
1. open cli and use node 10
2. run `npm install`
3. run `gulp` to build the change (then there will be changes built to dist/ folder automatically)

**Review by yidingww — CHANGES_REQUESTED** [review]
tested the functionality and it works now

however, need to make some adjustment to move the change at the proper place, see comment below

File: dist/select.js (hunk 1 above)
Comment (jperdiguerra) [reply dist/select.js:865]: ah I see, thanks!

**Review by yidingww — APPROVED** [review]
LGTM

**Comment by yidingww** [comment]
Let's create a PR in kaligo-web, and point to this branch by doing this in package.json. 

```json
"ui-select": "git+https://github.com/Kaligo/ui-select#bugfix/TE-3514-fields-changes-when-tabbing-out"
```

Then merge that PR to staging so the QA can test. Then after the PR in kaligo-web is signed off, let's merge this current PR and release a new tag and update the kaligo-web PR to point to the proper tag release

**Comment by jperdiguerra** [comment]
@playerace Can we merge this and add a release tag?

**Pull request merged**